crates/subspace-farmer/src/plot/piece_index_hash_to_offset_db.rs

```diff
@@ -61,6 +61,7 @@ impl IndexHashToOffsetDB {
         let mut options = Options::default();
         options.create_if_missing(true);
         options.create_missing_column_families(true);
+        options.set_use_fsync(true);
```

Comment:
This shouldn't be necessary, `fdatasync` is already used by RocksDB by default. Also comment on Rust version of RocksDB library is out of date: https://github.com/facebook/rocksdb/commit/65cd6cd4b62cbcb8f234bc3d6d25b7241b3ee55c

What we need instead is graceful shutdown support.

Reply:
Like the idea of graceful shutdown support, however, I wonder, how sending sigterm could lead to a corrupt state, since fdatasync is only different from fsync in that fdatasync doesn't make sure modification time is written.

Reply:
Without graceful shutdown we don't give underlying database to finish whatever it was doing and can potentially leave database in inconsistent state. Especially because database is written in C++ and its destructors can race with Rust's process shutdown if not handled carefully. Either way shutting it down gracefully should solve this.